File: frontend/lib/api/task.ts
import axiosInstance from '../axios';

export interface Task {
  id: number;
  title: string;
  description: string;
  documentId?: number;
  assigneeId: number;
  assigneeName?: string;
  status: 'TODO' | 'DOING' | 'DONE';
  dueDate?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTaskRequest {
  title: string;
  description: string;
  documentId?: number;
  assigneeId: number;
  dueDate?: string;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string;
  status?: 'TODO' | 'DOING' | 'DONE';
  dueDate?: string;
}

export interface TaskVO extends Task {
  assigneeName: string;
  documentTitle?: string;
}

/**
 * 创建任务
 */
export const createTask = async (data: CreateTaskRequest): Promise<TaskVO> => {
  const response = await axiosInstance.post('/api/tasks', data);
  return response.data.data;
};

/**
 * 获取我的任务列表
 */
export const getMyTasks = async (status?: string): Promise<TaskVO[]> => {
  const response = await axiosInstance.get('/api/tasks/my', {
    params: { status },
  });
  return response.data.data;
};

/**
 * 获取任务详情
 */
export const getTaskById = async (id: number): Promise<TaskVO> => {
  const response = await axiosInstance.get(`/api/tasks/${id}`);
  return response.data.data;
};

/**
 * 更新任务
 */
export const updateTask = async (id: number, data: UpdateTaskRequest): Promise<TaskVO> => {
  const response = await axiosInstance.put(`/api/tasks/${id}`, data);
  return response.data.data;
};

/**
 * 删除任务
 */
export const deleteTask = async (id: number): Promise<void> => {
  await axiosInstance.delete(`/api/tasks/${id}`);
};

File: backend/src/main/java/com/example/backend/entity/Task.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 任务实体类
 * 对应数据库表 tasks
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "tasks")
public class Task {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 关联的文档
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "document_id", nullable = false)
    private Document document;
    
    /**
     * 任务创建者
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;
    
    /**
     * 任务分配人
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assignee_id")
    private User assignee;
    
    /**
     * 任务标题
     */
    @Column(nullable = false, length = 255)
    private String title;
    
    /**
     * 任务描述
     */
    @Column(columnDefinition = "TEXT")
    private String description;
    
    /**
     * 任务状态: todo / doing / done
     */
    @Column(length = 20)
    @Builder.Default
    private String status = "todo";
    
    /**
     * 优先级: LOW / MEDIUM / HIGH
     */
    @Column(length = 20)
    @Builder.Default
    private String priority = "MEDIUM";
    
    /**
     * 截止日期
     */
    @Column(name = "due_date")
    private LocalDate dueDate;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
